modules/porous_flow/src/materials/PorousFlow1PhaseMD_Gaussian.C

```diff
@@ -0,0 +1,168 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "PorousFlow1PhaseMD_Gaussian.h"
+
+template<>
+InputParameters validParams<PorousFlow1PhaseMD_Gaussian>()
+{
+  InputParameters params = validParams<PorousFlowVariableBase>();
+  params.addRequiredCoupledVar("mass_density", "Variable that represents log(mass-density) of the single phase");
+  params.addRequiredRangeCheckedParam<Real>("al", "al>0", "For this class, the capillary function is assumed to be saturation = exp(-(al*porepressure)^2) for porepressure<0.");
+  params.addRequiredRangeCheckedParam<Real>("density_P0", "density_P0>0", "The density of the fluid phase at zero porepressure");
+  params.addRequiredRangeCheckedParam<Real>("bulk_modulus", "bulk_modulus>0", "The constant bulk modulus of the fluid phase");
+  params.addClassDescription("This Material is used for the single-phase situation where log(mass-density) is the primary variable.  calculates the 1 porepressure and the 1 saturation in a 1-phase isothermal situation, and derivatives of these with respect to the PorousFlowVariables.  A gaussian capillary function is assumed");
+  return params;
+}
+
+PorousFlow1PhaseMD_Gaussian::PorousFlow1PhaseMD_Gaussian(const InputParameters & parameters) :
+    PorousFlowVariableBase(parameters),
+
+    _al(getParam<Real>("al")),
+    _al2(std::pow(_al, 2)),
+    _logdens0(std::log(getParam<Real>("density_P0"))),
+    _bulk(getParam<Real>("bulk_modulus")),
+    _recip_bulk(1.0/_al/_bulk),
+    _recip_bulk2(std::pow(_recip_bulk, 2)),
+
+    _md_nodal_var(coupledNodalValue("mass_density")),
+    _md_qp_var(coupledValue("mass_density")),
+    _gradmd_qp_var(coupledGradient("mass_density")),
+    _md_varnum(coupled("mass_density"))
+{
+  if (_dictator_UO.numPhases() != 1)
+    mooseError("The Dictator proclaims that the number of phases is " << _dictator_UO.numPhases() << " whereas PorousFlow1PhaseMD_Gaussian can only be used for 1-phase simulations.  Be aware that the Dictator has noted your mistake.");
+}
+
+void
+PorousFlow1PhaseMD_Gaussian::initQpStatefulProperties()
+{
+  PorousFlowVariableBase::initQpStatefulProperties();
+
+  /*
+   *  YAQI HACK !!
+   *
+   * I really just want to simply call
+   * buildPS();
+   * but i do it below in computeQpProperties
+   */
+  buildPS();
+}
+
+void
+PorousFlow1PhaseMD_Gaussian::computeQpProperties()
+{
+  PorousFlowVariableBase::computeQpProperties();
+
+  buildPS();
+
+  /*
+   *  YAQI HACK !!
+   *
+   * I really just want to simply call
+   * buildPS();
+   * from initQpStatefulProperties, but the Variables
+   * aren't initialised at that point so moose crashes
```

Comment:
The YaqiHacky things are commented out because they are currently unnecessary in almost all situations.  However, i kept the comments here because of the issue #6774.  In one of our tests i have to use the

if (_t_step == 1)

YaqiHacky idea because MOOSE crashes when i buildPS (or buildPPSS in other classes) from initQpStatefulProperties.

I can delete the comments if you like, and simply provide an example of #6774 when we've got our code fully merged.

Just tell me what you prefer.